api/src/services/payload.ts

```diff
@@ -291,9 +291,27 @@ export class PayloadService {
 	 * Knex returns `datetime` and `date` columns as Date.. This is wrong for date / datetime, as those
 	 * shouldn't return with time / timezone info respectively
 	 */
-	processDates(payloads: Partial<Record<string, any>>[], action: Action): Partial<Record<string, any>>[] {
+	processDates(
+		payloads: Partial<Record<string, any>>[],
+		action: Action,
+		aliasMap: Record<string, string> = {},
+	): Partial<Record<string, any>>[] {
 		const fieldsInCollection = Object.entries(this.schema.collections[this.collection]!.fields);
 
+		for (const alias in aliasMap) {
+			const field = fieldsInCollection.find(([name, _]) => name === aliasMap[alias]);
```

Comment:
In terms of performance an object lookup should be faster and since we use the key name it should not be an issue?

```suggestion
			const field = this.schema.collections[this.collection]!.fields[aliasMap[alias]];
```